app/workers/databases.php

```diff
@@ -151,9 +179,20 @@ protected function deleteAttribute(Document $database, Document $collection, Doc
         // - deleting: was available, in deletion queue for first time
         // - failed: attribute was never created
         // - stuck: attribute was available but cannot be removed
+
         try {
-            if ($status !== 'failed' && !$dbForProject->deleteAttribute('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $key)) {
-                throw new Exception('Failed to delete Attribute');
+            if ($status !== 'failed') {
+                switch ($type) {
+                    case Database::VAR_RELATIONSHIP:
+                        if (!$dbForProject->deleteRelationship('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $key)) {
+                            throw new Exception('Failed to delete Attribute');
+                        }
+                        break;
+                    default:
+                        if (!$dbForProject->deleteAttribute('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $key)) {
+                            throw new Exception('Failed to delete Attribute');
+                        }
+                }
```

Comment:
do we need a switch if its just 1 condition right now? maybe a simple if/else is sufficient